### PR TITLE
LINK-1888 | 1/4 Price group list page

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Page to list and search price groups

This is 1st of the PRs to implement price group admin pages. All PRs are:
- https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/344
- https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/345
- https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/346
- https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/347

## Partially closes
[LINK-1888](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1888)

## Screenshot
Screenshot 2024-03-03 at 22 08 41

[LINK-1888]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1888?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/app/i18n/en.json [hunk 1]
```diff
@@ -1438,6 +1442,7 @@
     "permission": "permission",
     "place": "place",
     "platform": "platform",
+    "priceGroup": "price, group",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/en.json:1445]:
```suggestion
    "priceGroup": "price group",
```
I suppose comma was not meant to be here?

File: src/domain/app/i18n/fi.json [hunk 2]
```diff
@@ -1716,6 +1721,44 @@
     "warningNoRightsToCreate": "Sinulla ei ole oikeuksia luoda paikkoja.",
     "warningNoRightsToEdit": "Sinulla ei ole oikeuksia muokata tätä paikkaa."
   },
+  "priceGroup": {
+    "deletePriceGroupModal": {
+      "buttonDelete": "Poista hintaryhmä",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1726]: I think previously we decided to use the term "asiakasryhmä" in all three languages. Probably should be used in the administration UI too.

**Review by harriris-vincit — CHANGES_REQUESTED** [review]
Seems to have a bug that causes an empty page to occur when first searching price groups with a non-existent value (I used the string "agg") and then removing the search term.

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/app/i18n/en.json (hunk 1 above)
Comment (jorilindell) [reply src/domain/app/i18n/en.json]: Changed to `customer group` to unify the terms

File: src/domain/app/i18n/fi.json (hunk 2 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Replaced "hintaryhmä" with the term "asiakasryhmä" in all three languages.

**Comment by jorilindell** [comment]
> Seems to have a bug that causes an empty page to occur when first searching price groups with a non-existent value (I used the string "agg") and then removing the search term.

Fixed the bug in this PR as well. PR can now be re-reviewed.

**Review by harriris-vincit — APPROVED** [review]
Huge PR, but LGTM 👍

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr344-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr344-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**